include/deal.II/fe/fe_nothing.h

```diff
@@ -147,57 +147,39 @@ class FE_Nothing : public FiniteElement<dim,spacedim>
 
-  /**
-   * Fill the fields of FEValues. This function performs all the operations
-   * needed to compute the data of an FEValues object.
-   *
-   * In the current case, this function returns no meaningful information,
-   * since the element has no degrees of freedom.
-   */
   virtual
   void
-  fill_fe_values (const Mapping<dim,spacedim> &mapping,
+  fill_fe_values (const Mapping<dim,spacedim>                               &mapping,
                   const typename Triangulation<dim,spacedim>::cell_iterator &cell,
-                  const Quadrature<dim> &quadrature,
-                  const typename Mapping<dim,spacedim>::InternalDataBase &mapping_data,
-                  const typename Mapping<dim,spacedim>::InternalDataBase &fedata,
-                  FEValuesData<dim,spacedim> &data,
-                  const CellSimilarity::Similarity cell_similarity) const;
+                  const Quadrature<dim>                                     &quadrature,
+                  const typename Mapping<dim,spacedim>::InternalDataBase    &mapping_internal,
+                  const typename Mapping<dim,spacedim>::InternalDataBase    &fe_internal,
+                  const internal::FEValues::MappingRelatedData<dim,spacedim> &mapping_data,
+                  internal::FEValues::FiniteElementRelatedData<dim,spacedim> &output_data,
+                  const CellSimilarity::Similarity                           cell_similarity) const;
 
-  /**
-   * Fill the fields of FEFaceValues. This function performs all the
-   * operations needed to compute the data of an FEFaceValues object.
-   *
-   * In the current case, this function returns no meaningful information,
-   * since the element has no degrees of freedom.
-   */
   virtual
   void
-  fill_fe_face_values (const Mapping<dim,spacedim> &mapping,
+  fill_fe_face_values (const Mapping<dim,spacedim>                               &mapping,
                        const typename Triangulation<dim,spacedim>::cell_iterator &cell,
-                       const unsigned int face,
-                       const Quadrature<dim-1> & quadrature,
-                       const typename Mapping<dim,spacedim>::InternalDataBase &mapping_data,
-                       const typename Mapping<dim,spacedim>::InternalDataBase &fedata,
-                       FEValuesData<dim,spacedim> &data) const;
+                       const unsigned int                                         face_no,
+                       const Quadrature<dim-1>                                   &quadrature,
+                       const typename Mapping<dim,spacedim>::InternalDataBase    &mapping_internal,
+                       const typename Mapping<dim,spacedim>::InternalDataBase    &fe_internal,
+                       const internal::FEValues::MappingRelatedData<dim,spacedim> &mapping_data,
+                       internal::FEValues::FiniteElementRelatedData<dim,spacedim> &output_data) const;
```

Comment:
I plan on resorting the argument list in one of my next patches. The indentation will be fixed on that occasion.